src/lib/es2015.proxy.d.ts

```diff
@@ -1,21 +1,114 @@
 interface ProxyHandler<T extends object> {
+    /**
+     * A trap method for a function call.
+     * @param target The target callable object.
+     * @param thisArg The `this` argument for the call.
+     * @param argArray The list of arguments for rthe call
+     */
     apply?(target: T, thisArg: any, argArray: any[]): any;
+
+    /**
+     * A trap for the `new` operator.
+     * @param target The target object.
+     * @param argArray The list of arguments for the constructor.

```

Comment:
I would much rather have a longer doc comment and few comments per parameter. Most of the parameter comments are either trivial as-is, or refer to something that's not defined and would be trivial if there were a good explanation. The reference to `constructor` is a good example of the latter.

Reply:
Thank you for the feedback! I have made some changes, please let me know if they are adequate.

Should the trivial short param comments be removed altogether?

Reply:
Yes, I think so. I'll point out the ones that feel trivial to me and you can let me know what you think.

Reply:
Thank you! I have removed the trivial comments (I agree with all of them being trivial 😃) and added some clearer variable names.